Modules/Microsoft365DSC/DSCResources/MSFT_TeamsTenantDialPlan/MSFT_TeamsTenantDialPlan.psm1

```diff
@@ -195,6 +195,8 @@ function Set-TargetResource
     #endregion
 
     $CurrentValues = Get-TargetResource @PSBoundParameters
+    $PSBoundParameters = Remove-M365DSCAuthenticationParameter -BoundParameters $PSBoundParameters
+    $PSBoundParameters.Remove('OptimizeDeviceDialing') | Out-Null
```

Comment:
I believe this parameter is not deprecated. If so, we should indicate it as a comment in code signature of each of the Get/Test/Set method that it is deprecated, and we should print out a warning from the SET to the users letting them know (if they specify it) that it won't have any effect.

Reply:
As per https://learn.microsoft.com/en-us/powershell/module/skype/set-cstenantdialplan?view=skype-ps#notes, it is deprecated. I'll add a Write-Verbose warning.

The ExternalAccessPrefix and OptimizeDeviceDialing parameters have been removed from New-CsTenantDialPlan and Set-CsTenantDialPlan cmdlet since they are no longer used. External access dialing is now handled implicitly using normalization rules of the dial plans. The Get-CsTenantDialPlan will still show the external access prefix in the form of a normalization rule of the dial plan.

Reply:
@nikcharlebois Done, I've also issued a warning for ExternalAccessPrefix since that one is also deprecated.